File: Source/ObjC/NSKDefaultImageLayout.m
//
//  NSKDefaultImageLayout.m
//  NSKButtonObjC
//
//  Created by NSSimpleApps on 21.08.16.
//  Copyright © 2016 NSSimpleApps. All rights reserved.
//

#import "NSKDefaultImageLayout.h"


@implementation NSKDefaultImageLayout

+ (CGRect)titleRectForContentRect:(CGRect)contentRect
                 defaultTitleRect:(CGRect)defaultTitleRect
                        imageRect:(CGRect)imageRect
                  titleEdgeInsets:(UIEdgeInsets)titleEdgeInsets {
    
    return defaultTitleRect;
}

+ (CGRect)imageRectForContentRect:(CGRect)contentRect
                 defaultImageRect:(CGRect)defaultImageRect
                        titleRect:(CGRect)titleRect
                  imageEdgeInsets:(UIEdgeInsets)imageEdgeInsets {
    
    return defaultImageRect;
}

+ (CGSize)intrinsicContentSizeForDefault:(CGSize)defaultIntrinsicContentSize
                               imageRect:(CGRect)imageRect
                               titleRect:(CGRect)titleRect {
    
    return defaultIntrinsicContentSize;
}

@end

File: Source/Swift/NSKDefaultImageLayout.swift
//
//  NSKDefaultImageLayout.swift
//  NSKButtonSwift
//
//  Created by NSSimpleApps on 24.08.16.
//  Copyright © 2016 NSSimpleApps. All rights reserved.
//

import UIKit

internal class NSKDefaultImageLayout {
    
    private init() { }
    
    class func titleRect(forContentRect contentRect: CGRect,
                         defaultTitleRect: CGRect,
                         imageRect: CGRect,
                         titleEdgeInsets: UIEdgeInsets) -> CGRect {
        return defaultTitleRect
    }
    
    class func imageRect(forContentRect contentRect: CGRect,
                         defaultImageRect: CGRect,
                         titleRect: CGRect,
                         imageEdgeInsets: UIEdgeInsets) -> CGRect {
        return defaultImageRect
    }
    
    class func intrinsicContentSize(forDefaultIntrinsicContentSize defaultIntrinsicContentSize: CGSize,
                                    imageRect: CGRect,
                                    titleRect: CGRect) -> CGSize {
        return defaultIntrinsicContentSize
    }
}
